### PR TITLE
add ingress labels

### LINKED ISSUE #573: Question: adding ingress metrics?
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind feature

**What happened**:

Hi. Wanted to check what the maintainers think about exposing ingress metrics. 

I saw that there was a PR (https://github.com/kubernetes/kube-state-metrics/pull/225) that was closed. 

In our case, the use case is to alert on possible duplicate ingress paths created for the same host (undefined behaviour on our system).

I can think of a number of other use cases, like alerting of exposed test paths on prod.

Let me know what you think, and if it's worth it to re-open that PR, or make a new one adding ingress metrics. 

Thanks

### PR DESCRIPTION
**Pull request by r0fls**
**What this PR does / why we need it**:
Add an Ingress collector. There is currently no ingress collector.

**Which issue(s) this PR fixes**:
Fixes #573

### CODE REVIEW

File: pkg/collectors/builder.go [hunk 1]
```diff
@@ -21,6 +21,8 @@ import (
 	"sort"
 	"strings"
 
+    "k8s.io/api/extensions/v1beta1"
```
Comment (mxinden) [inline pkg/collectors/builder.go]: Indentation

File: pkg/collectors/ingress.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2018 The Kubernetes Authors All rights reserved.
```
Comment (mxinden) [inline pkg/collectors/ingress.go]:
```suggestion
Copyright 2019 The Kubernetes Authors All rights reserved.
```

File: pkg/collectors/ingress.go [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package collectors
+
+import (
+	"k8s.io/kube-state-metrics/pkg/metrics"
+
+    "k8s.io/api/extensions/v1beta1"
```
Comment (mxinden) [inline pkg/collectors/ingress.go]: Indentation.

File: pkg/collectors/ingress_test.go [hunk 4]
```diff
@@ -0,0 +1,107 @@
+/*
+Copyright 2018 The Kubernetes Authors All rights reserved.
```
Comment (mxinden) [inline pkg/collectors/ingress_test.go]:
```suggestion
Copyright 2019 The Kubernetes Authors All rights reserved.
```

File: pkg/collectors/ingress_test.go [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package collectors
+
+import (
+	"testing"
+
+    "k8s.io/api/extensions/v1beta1"
```
Comment (mxinden) [inline pkg/collectors/ingress_test.go]: Indentation.

File: pkg/collectors/ingress_test.go [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+
+func TestIngressCollector(t *testing.T) {
+	// Fixed metadata on type and help text. We prepend this to every expected
+	// output so we only have to modify a single place when doing adjustments.
+
+	startTime := 1501569018
+	metav1beta1StartTime := metav1.Unix(int64(startTime), 0)
+
+	const metadata = `
+        # HELP kube_ingress_labels Kubernetes labels converted to Prometheus labels.
```
Comment (mxinden) [inline pkg/collectors/ingress_test.go]: Indentation.

File: pkg/collectors/ingress_test.go [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+	// Fixed metadata on type and help text. We prepend this to every expected
+	// output so we only have to modify a single place when doing adjustments.
+
+	startTime := 1501569018
+	metav1beta1StartTime := metav1.Unix(int64(startTime), 0)
+
+	const metadata = `
+		# HELP kube_ingress_labels Kubernetes labels converted to Prometheus labels.
+		# TYPE kube_ingress_labels gauge
+        # HELP kube_ingress_info Information about ingress.
```
Comment (brancz) [inline pkg/collectors/ingress_test.go]: indentation if off here, seems like spaces instead of tabs like the other lines

**Comment by r0fls** [comment]
Anything else I can do here to help get this merged? I've tested using it on our clusters

**Comment by brancz** [comment]
this needs a rebase

**Comment by r0fls** [comment]
rebased :heavy_check_mark:

**Comment by r0fls** [comment]
@brancz rebased and added an ingress manifest for the e2e tests

**Comment by gaopeiliang** [comment]
Ingress info metrics not find in v1.5  ,  it is release ..   I want to find  ingress ==>> service info with metrics ....

**Pull request merged**